blog/2024-05-29-oras-1.2.0-and-oras-go-2.5.0/index.mdx

```diff
@@ -0,0 +1,98 @@
+## What's new in ORAS v1.2.0
+
+### OCI Spec v1.1.0 support
+
+ORAS CLI v1.2.0 and ORAS-go v2.5.0 are now compliant with OCI [image-spec v1.1.0](https://github.com/opencontainers/image-spec/blob/v1.1.0) and [distribution-spec v1.1.0](https://github.com/opencontainers/distribution-spec/tree/v1.1.0). 
+With OCI Spec v1.1.0 implemented by more and more OCI registries and officially supported in the ORAS client side, these major capabilities are officially enabled for users:
+
+- Able to create, store, and distribute non-container artifacts, such as Helm Chart, Kubernetes manifest file, See [OCI Artifact concept](https://oras.land/docs/concepts/artifact) for details.
+- Able to establish relationships between different artifacts. This allows users to associate the supply chain artifacts like SBOM, signature, vulnerability scanning report with the image. See [The Art of Associating Artifacts](https://oras.land/docs/concepts/reftypes#the-art-of-associating-artifacts) for details.
+- Able to discover and query artifact relationships, able to distribute a graph of artifacts across registries. See [Listing Referrers concept](https://oras.land/docs/concepts/reftypes#listing-referrers) for details.
+
+### Formatted output
+
+ORAS CLI has very basic output to show command operation result to human. For machine processing, especially in automation scenarios like scripting and CI/CD pipelines, developers may want to perform batch operations and chain different commands with ORAS, as well as filtering, modifying, and sorting objects based on the ORAS outputs.
+Developers expect that ORAS output can be emitted as machine-readable text, so that it can be used to perform further data manipulation.
+
+With formatted output support in ORAS v1.2.0, it enables users to use the `--format` to format metadata output into structured data (e.g. JSON) and optionally use the `--template` with the [Go template](https://pkg.go.dev/text/template) to manipulate the output data. 
+
+For example, push a file and two tags to a repository and show the descriptor of the image manifest in pretty JSON format:
+
+```bash
+oras push $REGISTRY/$REPO:$TAG1,$TAG2 sbom.spdx vul-scan.json --format json 
+```
+
+```json
+{
+  "reference": "$REGISTRY/$REPO@sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "digest": "sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "size": 820,
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  },
+  "artifactType": "json/example",
+  "referenceByTags": [
+    "$REGISTRY/$REPO:$TAG1",
+    "$REGISTRY/$REPO:$TAG2"
+  ]
+}
+```
+
+Futhermore, if you want to filter out the value of reference and mediaType of the pushed artifact in the standard output, use Go template functions as follows:
+
+```bash
+oras push $REGISTRY/$REPO:$TAG1,$TAG2 sbom.spdx vul-scan.json --format go-template='{{.reference}}, {{.mediaType}}'
```

Comment:
Would it be better to use `--template` flag for go-template in the  example as the `--template` flag is mainly designed for this purpose.

```suggestion
oras push $REGISTRY/$REPO:$TAG1,$TAG2 sbom.spdx vul-scan.json --format go-template --template '{{.reference}}, {{.mediaType}}'
```

Reply:
`--template` is designed to accommodate similar experience in `kubectl` and it's not recommended usage in ORAS. Better to avoid it in ORAS' documentations.

Reply:
@yizha1  The former experience is the recommended usage as it is simpler and will not be broken in the future. See the spec for reference: https://github.com/oras-project/oras/blob/main/docs/proposals/formatted-output.md#feature-status